### PR TITLE
fix: Static map open in chat when location has been searched

### LINKED ISSUE #1793: Not show maps in chat
**Actual Behaviour**

In the chat screen, not show maps 
**Expected Behaviour** 
Should be show maps

**Steps to reproduce it**
1. Go to chat
2. Try typing where is singapore?

**LogCat for the issue**

**Screenshots of the issue**

**Would you like to work on the issue?**

screenshot_2018-12-06-07-41-39

### PR DESCRIPTION
**Pull request by NileshPranami**
Fixes #1793 

 OpenStreetMap had stoped providing static map services
 Used token of mapBox for static map

### CODE REVIEW

**Comment by batbrain7** [comment]
@NileshPranami can you post a screenshot for the change?

**Comment by NileshPranami** [comment]
@batbrain7  Sir, have a look

screenshot_2019-01-15-23-10-56-501_ai susi

**Comment by batbrain7** [comment]
@NileshPranami is the token key your personal key ?

**Comment by batbrain7** [comment]
@iamareebjamal we need a fossasia key for this and also it should be stored in an environment variable?

**Comment by iamareebjamal** [comment]
Yes, it should not be hardcoded

**Comment by NileshPranami** [comment]
@iamareebjamal So please provide fossasia key so that I can make final change.

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/susi_android/pull/1841#issuecomment-454500760

**Comment by NileshPranami** [comment]
If Fossasia key is stored in environment variables then how can I use that. If I m missing something then please elaborate this.

**Comment by iamareebjamal** [comment]
Read about why environment variables are used. You'll put your key in environment variable. We'll put ours in our environment variable. You should not hardcode the API key in code. And you don't need access to our key for doing this. This is exactly why environment variables are used

**Comment by NileshPranami** [comment]
@iamareebjamal Sir, please review again my changes.

File: app/build.gradle [hunk 1]
```diff
@@ -7,6 +7,7 @@ apply plugin: 'kotlin-android-extensions'
 apply plugin: 'kotlin-kapt'
 apply plugin: 'realm-android'
 
+def STATIC_API_KEY = '"'+StaticTokenKey+'"' ?: '"Define your static api key"'
```
Comment (iamareebjamal) [inline app/build.gradle]: This is not using environment variable.

File: app/build.gradle [hunk 2]
```diff
@@ -34,10 +35,12 @@ android {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
             buildConfigField "String", "SPEAKER_BASE_URL", '"http://10.0.0.1:5000"'
+            buildConfigField "String", "Static_Map_Api", STATIC_API_KEY
```
Comment (iamareebjamal) [inline app/build.gradle]: What kind of naming convention is this? It's neither snake case, nor camel case

File: app/src/main/res/drawable/ic_share_white_24dp.xml [hunk 3]
```diff
@@ -4,6 +4,6 @@
     android:viewportHeight="24.0"
     android:viewportWidth="24.0">
     <path
-        android:fillColor="@color/md_white_1000"
+        android:fillColor="@color/md_grey_800"
```
Comment (iamareebjamal) [inline app/src/main/res/drawable/ic_share_white_24dp.xml]: Unneeded change

File: gradle.properties [hunk 4]
```diff
@@ -13,4 +13,6 @@
 
  org.gradle.parallel=true
 
- org.gradle.configureondemand=false
\ No newline at end of file
+ org.gradle.configureondemand=false
+
+ StaticTokenKey = pk.eyJ1IjoibmlsMTIzIiwiYSI6ImNqcXVkMWl1aDA2dmc0Mm10ZjBiYjJ1ZnUifQ.C7pKaawp7pV4uG9rpY9GjQ
```
Comment (iamareebjamal) [inline gradle.properties]: Unnecessary

File: app/build.gradle (hunk 1 above)
Comment (NileshPranami) [reply app/build.gradle]: @iamareebjamal Sir, Shall I use System.getenv() ?

File: app/src/main/res/drawable/ic_share_white_24dp.xml (hunk 3 above)
Comment (NileshPranami) [reply app/src/main/res/drawable/ic_share_white_24dp.xml]: Icon wasn't visible in white therefor I made change.

File: app/build.gradle (hunk 1 above)
Comment (iamareebjamal) [reply app/build.gradle]: yes

File: app/src/main/res/drawable/ic_share_white_24dp.xml (hunk 3 above)
Comment (NileshPranami) [reply app/src/main/res/drawable/ic_share_white_24dp.xml]: @iamareebjamal Sir, what to do about this, shall I remove it or consider it ?
When you long press on chat message then share icon will be visible.

Comment (iamareebjamal) [reply app/src/main/res/drawable/ic_share_white_24dp.xml]: Remove it

File: app/build.gradle [hunk 5]
```diff
@@ -7,8 +7,8 @@ apply plugin: 'kotlin-android-extensions'
 apply plugin: 'kotlin-kapt'
 apply plugin: 'realm-android'
 
-def STATIC_API_KEY = '"'+StaticTokenKey+'"' ?: '"Define your static api key"'
 def YOUTUBE_API_KEY = System.getenv('YOUTUBE_API_KEY') ?: "YOUR_API_KEY"
+def STATIC_API_KEY = '"'+System.getenv('MY_TOKEN')+'"' ?: '"DEFAULT"'
```
Comment (iamareebjamal) [inline app/build.gradle]: What's my token?

Comment (NileshPranami) [reply app/build.gradle]: Token's Variable name in system environment varaibles

Comment (iamareebjamal) [reply app/build.gradle]: What should I put in `MY_TOKEN` environment variable?

Comment (NileshPranami) [reply app/build.gradle]: Token for Static map generated by any admin of Fossasia organization.

Comment (iamareebjamal) [reply app/build.gradle]: So it should be named GOOGLE_MAPS_API_KEY ?

**Comment by NileshPranami** [comment]
Yes we can use Google static map API key but for that I have to change format little bit

File: app/build.gradle (hunk 5 above)
Comment (NileshPranami) [reply app/build.gradle]: It is not generated from Google. I use service of mapbox for static map.

Comment (NileshPranami) [reply app/build.gradle]: But if you say I can set format for Google static Map API

Comment (iamareebjamal) [reply app/build.gradle]: Change this to MAPBOX_API_KEY then

**Comment by iamareebjamal** [comment]
No, see the reason of failure and fix it. New PR will fail as well

File: gradle.properties [hunk 6]
```diff
@@ -13,4 +13,5 @@
 
  org.gradle.parallel=true
 
- org.gradle.configureondemand=false
\ No newline at end of file
+ org.gradle.configureondemand=false
+
```
Comment (batbrain7) [inline gradle.properties]: unnecessary changes
